java/src/jmri/jmrix/mqtt/MqttAdapter.java

```diff
@@ -69,7 +74,12 @@ public void connect() throws IOException {
             if (clientID.length() > 23) {
                 clientID = clientID.substring(clientID.length() - 23);
             }
-            mqttClient = new MqttClient(PROTOCOL + getCurrentPortName(), clientID);
+            String tempdirName = jmri.util.FileUtil.getExternalFilename("profile:");
```

Comment:
There is a public static for `"profile:"` in FileUtil (`jmri.util.FileUtil.PROFILE`) that should probably be used instead of a magic constant.

Reply:
Fixed, thanks.